pkg/infradb/bridge.go

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (c) 2022-2023 Dell Inc, or its subsidiaries.
+// Copyright (c) 2022-2023 Intel Corporation, or its subsidiaries.
+// Copyright (C) 2023 Nordix Foundation.
+
+package infradb
+
+import (
+	// "encoding/binary"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"net"
+
+	pb "github.com/opiproject/opi-api/network/evpn-gw/v1alpha1/gen/go"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/common"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/subscriberframework/eventbus"
+)
+
+// LogicalBridgeOperStatus operational Status for Logical Bridges
+type LogicalBridgeOperStatus int32
+
+const (
+	// LogicalBridgeOperStatusUnspecified for Logical Bridge unknown state
+	LogicalBridgeOperStatusUnspecified LogicalBridgeOperStatus = iota
+	// LogicalBridgeOperStatusUp for Logical Bridge up state
+	LogicalBridgeOperStatusUp = iota
+	// LogicalBridgeOperStatusDown for Logical Bridge down state
+	LogicalBridgeOperStatusDown = iota
+	// LogicalBridgeOperStatusToBeDeleted for Logical Bridge to be deleted state
+	LogicalBridgeOperStatusToBeDeleted = iota
+)
+
+// LogicalBridgeStatus holds Logical Bridge Status
+type LogicalBridgeStatus struct {
+	LBOperStatus LogicalBridgeOperStatus
+	Components   []common.Component
+}
+
+// LogicalBridgeSpec holds Logical Bridge Spec
+type LogicalBridgeSpec struct {
+	VlanID uint32
+	Vni    *uint32
+	VtepIP *net.IPNet
+}
+
+// LogicalBridgeMetadata holds Logical Bridge Metadata
+type LogicalBridgeMetadata struct{}
+
+// LogicalBridge holds Logical Bridge info
+type LogicalBridge struct {
+	Name            string
+	Spec            *LogicalBridgeSpec
+	Status          *LogicalBridgeStatus
+	Metadata        *LogicalBridgeMetadata
+	Svi             string
+	BridgePorts     map[string]bool
+	MacTable        map[string]string
+	ResourceVersion string
+}
+
+// build time check that struct implements interface
+var _ EvpnObject[*pb.LogicalBridge] = (*LogicalBridge)(nil)
+
+// NewLogicalBridge creates new Logica Bridge object from protobuf message
+func NewLogicalBridge(in *pb.LogicalBridge) *LogicalBridge {
+	var vip *net.IPNet
+	components := make([]common.Component, 0)
+
+	// Parse vtep IP
+	if in.Spec.VtepIpPrefix != nil {
+		vtepip := make(net.IP, 4)
+		binary.BigEndian.PutUint32(vtepip, in.Spec.VtepIpPrefix.Addr.GetV4Addr())
+		vip = &net.IPNet{IP: vtepip, Mask: net.CIDRMask(int(in.Spec.VtepIpPrefix.Len), 32)}
+	}
+
+	subscribers := eventbus.EBus.GetSubscribers("logical-bridge")
+	if len(subscribers) == 0 {
+		log.Println("NewLogicalBridge(): No subscribers for Logical Bridge objects")
```

Comment:
I cannot understand if missing subscribers is an error here... If it is an error, we need to return here. If it is not, do we need to print, since it is not an error?
Here, we treat it as an error https://github.com/opiproject/opi-evpn-bridge/pull/377/files#r1562602607

Reply:
Yes I guess we missed that. @atulpatel261194 can you please return an error also in these cases here as we do in the other cases where we do not have subscribers ?

Reply:
@mardim91 , @artek-koltun  New functions for each object just returns the domain specific object. we handle the error in infra db if there is no subscribers while create, delete and update of perticular object.

Reply:
Done, fixed